nixos/tests/hardened.nix

```diff
@@ -43,83 +43,80 @@ import ./make-test.nix ({ pkgs, ...} : {
       };
     in
     ''
-      $machine->waitForUnit("multi-user.target");
+      import re
 
-      subtest "apparmor-loaded", sub {
-          $machine->succeed("systemctl status apparmor.service");
-      };
+      machine.wait_for_unit("multi-user.target")
+
+      with subtest("apparmor-loaded"):
+          machine.succeed("systemctl status apparmor.service")
 
       # AppArmor securityfs
-      subtest "apparmor-securityfs", sub {
-          $machine->succeed("mountpoint -q /sys/kernel/security");
-          $machine->succeed("cat /sys/kernel/security/apparmor/profiles");
-      };
+      with subtest("apparmor-securityfs"):
+          machine.succeed("mountpoint -q /sys/kernel/security")
+          machine.succeed("cat /sys/kernel/security/apparmor/profiles")
 
       # Test loading out-of-tree modules
-      subtest "extra-module-packages", sub {
-          $machine->succeed("grep -Fq wireguard /proc/modules");
-      };
+      with subtest("extra-module-packages"):
+          machine.succeed("grep -Fq wireguard /proc/modules")
 
       # Test hidepid
-      subtest "hidepid", sub {
-          $machine->succeed("grep -Fq hidepid=2 /proc/mounts");
+      with subtest("hidepid"):
```

Comment:
it would be cool if all subtest titles would be at least short sentences that describe your high-level expectation, instead of containing just one word.